### DISCUSSION TITLE
Force Royalties + make contract upgradable/updateable

### DISCUSSION THREAD

**Discussion by ClydeMaster** [hashlips-lab/nft-erc721-collection#465] [Q&A]
Hi everyone!
As OpenSea cut off royalties for secondary sales, I would like to move away from OpenSea make sure I will receive royalties respectively.
Many questions/thoughts arise. Some info in advance:
- I use the ERC721 contract + Dapp from Hashlips (v 2.4.5)
- Collection is shown on Opensea, only about 2% of my NFTs have been minted through Whitelist, still hidden, 14 owners
- Chain: Polygon

Questions/ thoughts:
1. The contract is not updateable, is it?

2. If not updateable, I guess I have to deploy a new contract. Is there a function to include in the contract to make sure I will 
    receive royalties from secondary sales? Guess, will have to **block sales** on certain marketplaces that do not offer royalties  
    (Opensea, Blur,...)?

3. Or is it better to only **allow sales** on a special marketplace like Rarible (as they guarantee royalties)? What would I have to 
     change in the contract/Dapp then? 

4. What would I have to do to be able to update the contract after deployment in case some future changes will have to be 
     done? Such insecure times where e.g. marketplaces can do whatever they want with even already running projects might 
     need us to update our contracts at some stages.

5. My plan so far: 
    - Leave sale closed on current project. 
    - Deploy new contract (different name + of course, new contract address)
    - Drop NFTs from the new contract into the wallets of my current NFT holders. There are only 14 holders, so should not be a 
       lot of effort.
   Does this make sense or is there a better way/solution?

6. Do I have to contact Polygon support to delete my old contract or can I just leave it on the blockchain (sale 
     will stay closed for all times)?

7. There is no way to delete the current collection site on Opensea as I am not the only owner of the current NFTs, is there?

In https://github.com/hashlips-lab/nft-erc721-collection/issues/12 there was a similar question but could not find an answer there and it was related to Opensea royalties which are obsolete now.

OR is this a too complicated topic here and I should contact the OpenDevs Crew Discord? Am happy to buy a NFT, if I will receive a solution there. :-)

Thanks a lot in advance! Pls excuse the long text. Maybe it is interesting for some other readers/users too.

**Comment by liarco** [comment]
Hi @ClydeMaster, all the things you mentioned here make a lot of sense (and we are currently working hard to find good solutions to these issues). I knew the Operator Filter Registry could not be a solid solution in the long run, that's why I've been so against it's adoption as a default solution for this repo.

This topic can definitely be covered under the support provided to [OpenDevsCrew](https:///opendevs.io) holders so feel free to join us and I'll be more than happy to guide you through the solutions we tested so far.

I also wanna say I really appreciate you for mentioning ODC and your willingness to support our work by joining the Crew. 🙏

**Reply by lizaralmighty** [reply]
hii every one, opensea depreciated operator filter registry, could you delete that code in your smarcontract? and make function safetransform from and others in filter registry general not openseafilter registry again. thanks before.

**Reply by liarco** [reply]
Hi @lizaralmighty, our repo has never supported the Operator Filter Registry because I didn't like the implementation and the overall concept, so you don't have to remove anything.
The implementation discussed here is a custom one and it's private for ODC holders only, so ClydeMaster is already testing it for one of his collections, but we don't have an ETA for a public release yet.

If you wanna get early-access to it to test it and contribute to it's development you can join the OpenDevsCrew by minting a token and contact us in our private Discord channel.
